tools/manually-handling-sharding-ddl-locks.md

```diff
@@ -0,0 +1,415 @@
+---
+title: Handle Sharding DDL Locks Manually in TiDB-DM
+summary: Learn how to handle sharding DDL locks manually in TiDB-DM.
+category: tools
+---
+
+# Handle Sharding DDL Locks Manually in TiDB-DM
+
+The sharding DDL lock of TiDB-DM automatically completes the synchronization process in most cases, but under some abnormal scenarios you need the command `unlock-ddl-lock` or `break-ddl-lock` to handle abnormal DDL locks manually.
+
+> **Warning:**
+>- Do not use `unlock-ddl-lock` or `break-ddl-lock` unless you are totally aware of the possible impacts brought by the command and you can accept them.
+>- Before manually handling the abnormal DDL lock, make sure that you have already read the DM [shard merge principles](./shard-merge.md#Principles).
+
+### Command
+
+#### show-ddl-locks
+
+Show the current DDL lock information on `DM-master`.
+
+##### Command usage
+
+```bash
+show-ddl-locks [--worker=127.0.0.1:8262] [task-name]
+```
+
+##### Variables description
+
+- `worker`: flag; string; `--worker`; optional; can be specified multiple times; if not being specified, show the lock information of all DM-workers; if being specified, show the lock information of the specified DM-worker.
+- `task-name`: non-flag; string; optional; if not being specified, show the lock information of all tasks; if being specified, show the lock information of the specified task.
+
+##### Example of results
+
+```bash
+» show-ddl-locks test
+{
+    "result": true,                                        # show if the locking process succeeds
+    "msg": "",                                             # show the reason for the locking process failure or other descriptive information (for example, the locking task does not exist)
```

Comment:
the reason :thinking:  is `additional message` better?

Reply:
Sounds fair, I'll change it.